### DISCUSSION TITLE
How to read WKB geometries supplied as hex strings

### DISCUSSION THREAD

**Discussion by ghost** [duckdb/duckdb#12557] [Q&A] [answered]
I have installed and loaded the `spatial` extension into DuckDb 1.0.0. In PostgreSQL I am able to do, e.g.,
```
SELECT '0101000020E610000058C51B99471E2A40EE79FEB451334A40'::geometry;
SELECT ST_GeomFromEWKB(decode('0101000020E610000058C51B99471E2A40EE79FEB451334A40', 'hex'));
```
and both work (the input string is a point). What is the DuckDb equivalent? If I try the first one I get this error:
```
ConversionException: Conversion Error: WKT Parser: Unknown geometry type '0101000020E610000058C51B99471E2A40EE79FEB451334A40' at position 0 near: '0'|<---
```
and as to the other one neither `decode` nor `ST_GeomFromEWKB` seem to exist in DuckDb, although `ST_GeomFromWKB` does but using it with that string gives a warning in PostgreSQL.

**Comment by Maxxen** [comment, marked as answer]
Hi! There's a ST_GeomFromHEXWKB that should work! In DuckDB It handles both WKB and EWKB.

**Reply by ghost** [reply]
Yes, that worked. Thank you so much!
